frontend/src/metabase-lib/parameters/utils/parameter-values.js

```diff
@@ -9,12 +9,39 @@ import {
 export const PULSE_PARAM_EMPTY = null;
 export const PULSE_PARAM_USE_DEFAULT = undefined;
 
-export function getValuePopulatedParameters(parameters, parameterValues) {
+/**
+ * In some cases, we need to use default parameter value in place of an absent one.
+ * Please use this function when dealing with the required parameters.
+ */
+export function getParameterValue({
+  parameter,
+  values = {},
+  defaultRequired = false,
+}) {
+  const value = values?.[parameter.id];
```

Comment:
do you expect `null` in place of values? otherwise there is a default value `{}` and `?.` can be removed

Reply:
Yeah, you're right, it's some leftover.

Reply:
and tests failed 😄

Reply:
Yeah, this is why I had it in the first place — won't touch it now though.